lib/model/query/sessions.js

```diff
@@ -29,11 +29,27 @@ returning *`)
 
 const _unjoiner = unjoiner(Session, Actor);
 const getByBearerToken = (token) => ({ maybeOne }) => (isValidToken(token) ? maybeOne(sql`
-select ${_unjoiner.fields} from sessions
-join actors on actors.id=sessions."actorId"
-where token=${token} and sessions."expiresAt" > now()`)
+  SELECT "sessions"."csrf" AS "sessions!csrf"
+       , "actors"."id" AS "actors!id"
+       , "actors"."type" AS "actors!type"
+       , "actors"."acteeId" AS "actors!acteeId"
+    FROM sessions
+    JOIN actors ON actors.id=sessions."actorId"
+    WHERE token=${token}
+      AND sessions."expiresAt" > NOW()
+      AND actors."deletedAt" IS NULL
+`)
+  .then(map(auth => assoc('sessions!token', token, auth)))
   .then(map(_unjoiner)) : Promise.resolve(Option.none()));
 
+const getLifetime = (token) => ({ maybeOne }) => (isValidToken(token) ? maybeOne(sql`
+  SELECT "createdAt"
+       , "expiresAt"
+    FROM sessions
+    WHERE token=${token}
+      AND sessions."expiresAt" > NOW()
```

Comment:
Should we filter on `actors."deletedAt"` as well just for consistency with the other query above?

Reply:
That seems prudent - it wouldn't be good to restore a session for an actor that's been deleted.  Or is that an impossible scenario?

How about renaming the function from `getLifetime()` to `getValidLifetime()`?

Reply:
> Checking `actors."deletedAt"` here is unnecessary as `/restore` already calls `getOrNotFound(auth.session)`. Do you think joining on `actors` and checking `actors."deletedAt"` is still worthwhile?

That's a good point. If we know that the token passed to `getLifetime()` has been taken from a session whose existence/validity has been confirmed, I don't feel like we need to filter on `actors."deletedAt"`. What do you think about changing the argument of `getLifetime()` so that it takes a session object rather than a token? That'd be a way to prove to `getLifetime()` that we've already confirmed the existence/validity of the session and aren't just passing in any old token.

Reply:
Done.